src/store/validation/validationHelpers.ts

```diff
@@ -121,8 +121,15 @@ export function validateCommunityBlock({block, schemaVersion}: {block: IBlock, s
   let validate = null
   if (isEmpty(validators) || !validators.has(block.type)) {
     const blockTypeWithoutNameSpace = block.type.split('.')[block.type.split('.').length - 1]
-    const blockJsonSchema = require(`@floip/flow-runner/dist/resources/validationSchema/${schemaVersion}/I${blockTypeWithoutNameSpace}Block.json`)
-    validate = createDefaultJsonSchemaValidatorFactoryFor(blockJsonSchema)
+    let blockJsonSchema
+    try {
+      blockJsonSchema = require(`@floip/flow-runner/dist/resources/validationSchema/${schemaVersion}/I${blockTypeWithoutNameSpace}Block.json`)
+      validate = createDefaultJsonSchemaValidatorFactoryFor(blockJsonSchema)
+    } catch (e) {
```

Comment:
I believe it's actually the require that's failing here. I need to think about this a bit more and document a suggested way to add custom validators. For a third party block the schema is never going to be at this path (in flow-runner). I suspect for third party blocks we need to override the whole validation function in the store. In this function though, perhaps I should explicitly check that the file exists rather than trying to require it whatever?

Reply:
I'm not finding a good way of explicitly checking the module exists so I'm just catching and rethrowing if the error code isn't `MODULE_NOT_FOUND`. Let me know if you know a good solution

Reply:
Thanks for the improvement, much better I think